### PR TITLE
Test/core jax builder

### LINKED ISSUE #545: [TEST] Jax System Core Executor
### What do you want to test?
Jax system core executor (in `mava/systems/jax/`)

### Outline of test structure
* Unit tests
* Test components and hooks

### Definition of done
Passing checks, cover all hooks, edge cases considered

### Mandatory checklist before making a PR
* [ ] The success criteria laid down in “Definition of done” are met.
* [ ] Test code is documented - docstrings for methods and classes, static types for arguments.
* [ ] Documentation is updated - README, CONTRIBUTING, or other documentation.

### PR DESCRIPTION
**Pull request by mmorris44**
## What?
Test for core Mava executor.
## Why?
Robust testing for Mava release.
## How?
- Wrote unit tests for methods in the class
- Used `HookOrderTracking` to test hook orders
- Moved builder integration test out of existing file and into a new folder
## Extra
Closes #545. Tests pass and example runs.
**Note:** tests in the `integration` folder are not new, they're just moved from previous locations.

### CODE REVIEW

File: tests/jax/core_system_components/builder_test.py [hunk 1]
```diff
@@ -1,106 +1,223 @@
...
+    )
+
+    assert test_builder.store.executor_id == executor_id
+    assert test_builder.store.data_server_client == data_server_client
+    assert test_builder.store.parameter_server_client == parameter_server_client
+    assert test_builder.store.is_evaluator
+    assert test_builder.store.adder is None
+
+    assert isinstance(test_builder.store.executor, Executor)
+    assert test_builder.store.executor.store == test_builder.store
```
Comment (RuanJohn) [inline tests/jax/core_system_components/builder_test.py:73]: Can we also assert that the correct components were passed in? Same for in other places. Right now it should be an empty list everywhere but perhaps you could just have one component in the builder initially and check that it is passed through properly.

**Review by RuanJohn — COMMENTED** [review]
Very very neat and well structured PR @mmorris44. Just a really small note from my side. I also am not looking at the integration test since the goal here is just to move it.

**Review by AsadJeewa — CHANGES_REQUESTED** [review]
Thanks @mmorris44  minor changes requested

File: tests/jax/core_system_components/builder_test.py (hunk 1 above)
Comment (mmorris44) [reply tests/jax/core_system_components/builder_test.py:73]: Good idea, done!

File: tests/jax/core_system_components/builder_test.py [hunk 2]
```diff
@@ -1,106 +1,223 @@
...
+def test_data_server_store(test_builder: TestBuilder) -> None:
+    """Test that store is handled correctly in data_server()."""
+    assert test_builder.data_server() == ["data_table_1", "data_table_2"]
+
+
+def test_parameter_server_store(test_builder: TestBuilder) -> None:
+    """Test that store is handled correctly in parameter_server()."""
+    parameter_server = test_builder.parameter_server()
+    assert isinstance(parameter_server, ParameterServer)
+    assert parameter_server.store == test_builder.store
```
[the remark this answers is not in the document]
Comment (mmorris44) [reply tests/jax/core_system_components/builder_test.py:49]: Done.

File: tests/jax/core_system_components/builder_test.py [hunk 3]
```diff
@@ -1,106 +1,223 @@
...
+    assert not test_builder.store.is_evaluator
+    assert test_builder.store.adder == "adder"
+
+    assert isinstance(test_builder.store.executor, Executor)
+    assert test_builder.store.executor.store == test_builder.store
+
+
+def test_trainer_store(test_builder: TestBuilder) -> None:
+    """Test that store is handled correctly in trainer()."""
+    trainer_id = "executor"
```
[the remark this answers is not in the document]
Comment (mmorris44) [reply tests/jax/core_system_components/builder_test.py:102]: Nice catch! Fixed.

**Review by AsadJeewa — APPROVED** [review]
Thanks. Happy on my side
